test/suite/test_backup17.py

```diff
@@ -0,0 +1,165 @@
+                # 2 is WT_BACKUP_RANGE
+                self.assertTrue(curtype == 1 or curtype == 2)
+                if curtype == 1:
+                    self.pr('Copy from: ' + newfile + ' (' + str(size) + ') to ' + self.dir)
+                    shutil.copy(newfile, self.dir)
+                else:
+                    self.pr('Range copy file ' + newfile + ' offset ' + str(offset) + ' len ' + str(size))
+                    offsets.append(offset)
+                    rfp = open(newfile, "r+b")
+                    wfp = open(self.dir + '/' + newfile, "w+b")
+                    rfp.seek(offset, 0)
+                    wfp.seek(offset, 0)
+                    if size > self.granval:
+                        saw_multiple = True
+                    buf = rfp.read(size)
+                    wfp.write(buf)
+                    rfp.close()
+                    wfp.close()
+                dup_cnt += 1
+            dupc.close()
+        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
+        bkup_c.close()
+        if consolidate:
+            self.assertTrue(saw_multiple)
+        else:
+            self.assertFalse(saw_multiple)
+        return offsets
+
+    def test_backup17(self):
+
+        self.session.create(self.uri, "key_format=S,value_format=S")
+        self.session.create(self.uri2, "key_format=S,value_format=S")
+        self.add_data(self.uri)
+        self.add_data(self.uri2)
+        self.mult += 1
+
+        # Open up the backup cursor. This causes a new log file to be created.
+        # That log file is not part of the list returned. This is a full backup
+        # primary cursor with incremental configured.
+        os.mkdir(self.dir)
+        config = 'incremental=(enabled,granularity=%s,this_id="ID1")' % self.gran
+        bkup_c = self.session.open_cursor('backup:', None, config)
+
+        # Now copy the files returned by the backup cursor.
+        all_files = []
+        while True:
+            ret = bkup_c.next()
+            if ret != 0:
+                break
+            newfile = bkup_c.get_key()
+            sz = os.path.getsize(newfile)
+            self.pr('Copy from: ' + newfile + ' (' + str(sz) + ') to ' + self.dir)
+            shutil.copy(newfile, self.dir)
+            all_files.append(newfile)
+        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
+        bkup_c.close()
+
+        # This is the main part of the test for consolidate. Add data to the first table.
+        # Then perform the incremental backup with consolidate off (the default). Then add the
+        # same data to the second table. Perform an incremental backup with consolidate on and
+        # verify we get fewer, consolidated values.
+        self.add_data(self.uri)
+        uri1_off = self.take_incr_backup(2, False)
+
+        self.add_data(self.uri2)
+        uri2_off = self.take_incr_backup(3, True)
+
+        # Assert that we recorded fewer offsets on the consolidated backup.
+        self.assertLess(len(uri2_off), len(uri1_off))
```

Comment:
For some extra verification, it would be nice to return a list of sizes, rather than a list of offsets.  You can still verify the lengths, but you can make sure the totals are the same.  Helpful python function `sum`:
```
$ python
>>> sizes=[100, 200, 34]
>>> total=sum(sizes)
>>> total
334
>>> 
```

Reply:
Good idea. Fixed.